rfd/0120-cloud-agent-auto-upgrades-ux.md

```diff
@@ -0,0 +1,68 @@
+---
+authors: Xin Ding (xin@goteleport.com)
+state: draft
+---
+
+# RFD 120 - Cloud Agent Auto Upgrades UX
+
+## Required Approvals
+
+* Engineering: @jimbishopp && @r0mant
+* Product: @klizhentas
+* Security: @reedloden
+
+## What
+
+Teleport will add support for Agent Auto Upgrades in v13.0.0. This document
+details tactical UX changes to expose the Agent Auto Upgrade capabilities to
+Teleport Cloud Admins.
+
+### Terminology
+
+* (Teleport) Practitioner: People who are actively using Teleport in their
+  day-to-day.
+* (Teleport) Admins: A subset of Practitioners. Admins deploy, configure, and
+  otherwise set up Teleport.
+* (Teleport) End Users: A subset of Practitioners. End Users use Teleport to
+  access infrastructure resources.
+* (Teleport) Agent: The `teleport` process which can run one more more Teleport
+  services.
+* (Teleport) Cluster Alerts: Messages in the Teleport Web UI and appropriate
+  Teleport CLIs that alert Practitioners of relevant concerns.
+* Unaccompanied Agent: A Teleport Agent deployed without an accompanying auto upgrader
+  service.
+
+## Why
+
+Simply adding support for Agent Auto Upgrades doesn't help if Admins don't adopt
+it. We want all Teleport customers to enable Agent Auto Upgrades in order to
+reduce the significant workload associated with manually managing and upgrading
+a fleet of Teleport Agents.
+
+## How
```

Comment:
This section doesn't differentiate between tenants enrolled in automatic updates and tenants who are not. Even if in the end we want to enroll everyone, we might want to explain that this only applies to the tenants enrolled in this new mode.